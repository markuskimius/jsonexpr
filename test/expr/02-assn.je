a = 0xf0f; PRINT("a &= 0xff0");  PRINT(a &= 0xff0); PRINT(a);  PRINT();
a = 1024;  PRINT("a %= 7");      PRINT(a %= 7);     PRINT(a);  PRINT();
a = 1024;  PRINT("a *= 2");      PRINT(a *= 2);     PRINT(a);  PRINT();
a = 1024;  PRINT("a **= 2");     PRINT(a **= 2);    PRINT(a);  PRINT();
a = 1024;  PRINT("a += 2");      PRINT(a += 2);     PRINT(a);  PRINT();
a = 1024;  PRINT("a -= 2");      PRINT(a -= 2);     PRINT(a);  PRINT();
a = 1024;  PRINT("a /= 2");      PRINT(a /= 2);     PRINT(a);  PRINT();
a = 1024;  PRINT("a <<= 2");     PRINT(a <<= 2);    PRINT(a);  PRINT();
a = 1024;  PRINT("a >>= 2");     PRINT(a >>= 2);    PRINT(a);  PRINT();
a = 1024;  PRINT("a >>>= 2");    PRINT(a >>>= 2);   PRINT(a);  PRINT();
a = 0xf0f; PRINT("a ^= 0xff0");  PRINT(a ^= 0xff0); PRINT(a);  PRINT();
a = 0xf0f; PRINT("a |= 0xff0");  PRINT(a |= 0xff0); PRINT(a);  PRINT();

a = -1024; PRINT("a <<= 2");     PRINT(a <<= 2);    PRINT(a);  PRINT();
a = -1024; PRINT("a >>= 2");     PRINT(a >>= 2);    PRINT(a);  PRINT();
a = -1024; PRINT("a >>>= 2");    PRINT(a >>>= 2);   PRINT(a);  PRINT();
