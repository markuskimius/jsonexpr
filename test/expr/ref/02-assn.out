a &= 0xff0
3840
3840

a %= 7
2
2

a *= 2
2048
2048

a **= 2
1048576.0
1048576.0

a += 2
1026
1026

a -= 2
1022
1022

a /= 2
512
512

a <<= 2
4096
4096

a >>= 2
256
256

a >>>= 2
256
256

a ^= 0xff0
255
255

a |= 0xff0
4095
4095

a <<= 2
-4096
-4096

a >>= 2
-256
-256

a >>>= 2
4611686018427387648
4611686018427387648

